eng/Scripts/CI-Test.sh

```diff
@@ -19,4 +19,30 @@ if [ ! -f "$RootDir/bin/DebugAdapterProtocolTests/Debug/CppTests/config.xml" ];
     fi
 fi
 
-dotnet test "$RootDir"/bin/DebugAdapterProtocolTests/Debug/CppTests/CppTests.dll --logger "trx;LogFileName=$RootDir/bin/DebugAdapterProtocolTests/Debug/CppTests/results.trx"
\ No newline at end of file
+# Adapted from https://www.cazzulino.com/dotnet-test-retry.html
+counter=0
+exitcode=0
+while [ $counter -lt 6 ]
```

Comment:
>6 [](http://example.com/codeflow?start=21&length=1)

I don't think we should treat a test that passes 1 out of 6 runs as passed.

Would it be possible to instead use the logic from the C# extension? That will rerun failed tests, but:
* There is a maximum number of tests we will retry
-and-
* If a test fails initially, it needs to repeatedly pass on rerun (4 times is what the C# tests use)

See RetrySettings in /src/vsdbg/VsDbg-UITests/TestRunner/CustomTestCase.cs.